--- a/drivers/net/wireless/ath/ath9k/ath9k.h
+++ a/drivers/net/wireless/ath/ath9k/ath9k.h
@@ -136,10 +136,10 @@
 
 #define ATH_MAX_ANTENNA         3
 #define ATH_RXBUF               512
-#define ATH_TXBUF               512
+#define ATH_TXBUF               32
 #define ATH_TXBUF_RESERVE       5
 #define ATH_MAX_QDEPTH          (ATH_TXBUF / 4 - ATH_TXBUF_RESERVE)
-#define ATH_TXMAXTRY            13
+#define ATH_TXMAXTRY            4
 #define ATH_MGT_TXMAXTRY        4
 
 #define TID_TO_WME_AC(_tid)				\
@@ -193,7 +193,7 @@
 	ATH_AGGR_LIMITED,
 };
 
-#define ATH_TXFIFO_DEPTH 8
+#define ATH_TXFIFO_DEPTH 4
 struct ath_txq {
 	int axq_class;
 	u32 axq_qnum;
